TCMManagement/BusinessLayer/PatientService.cs

```diff
@@ -31,23 +31,34 @@ public IEnumerable<Patient> GetItems(IEnumerable<KeyValuePair<string, string>> q
         {
             if (!Utils.IsNullOrEmpty(queryParams))
             {
-                // only support get patient thru phone number, and firstname (1st) + lastname (2nd)
+                // only support get patient thru email, phone number, and firstname (1st) + lastname (2nd)
                 List<KeyValuePair<string, string>> queryList = queryParams.ToList();
-                if (queryParams.First().Key == "Phone")
+                if (queryParams.FirstOrDefault().Key == "Email")
                 {
-                    return context.Patients
+                    return (new List<Patient>{SearchItem(queryParams.FirstOrDefault().Value)});
+                }
+                if (queryParams.FirstOrDefault().Key == "Phone")
+                {
+                    // If I put queryParams.FirstOrDefault().Value to Where() directly, I will get exception
+                    string phoneNum = queryParams.FirstOrDefault().Value;

```

Comment:
Looks like can't put "queryParams.FirstOrDefault().Value" directly to context.Where otherwise will get exception.

Reply:
Get it.